ApiReview/API.Polly.Core/NoDocs/Polly.Strategy/ResilienceStrategyOptions.cs

```diff
@@ -0,0 +1,14 @@
+// Assembly 'Polly.Core'
+
+using System.ComponentModel.DataAnnotations;
+using System.Runtime.CompilerServices;
+
+namespace Polly.Strategy;
+
+public abstract class ResilienceStrategyOptions
+{
+    [Required(AllowEmptyStrings = true)]
+    public string StrategyName { get; set; }
+    public abstract string StrategyType { get; }
```

Comment:
What's this used for? Can't this be derived using GetType at runtime?

Reply:
`StrategyName` -> `InstanceName`, nullable but give it some default
`StrategyType` -> hide for now (we can introduce later if needed)